streams/src/test/java/org/apache/kafka/streams/processor/internals/StreamTaskTest.java

```diff
@@ -115,7 +109,8 @@
     private final byte[] recordKey = intSerializer.serialize(null, 1);
     private final String applicationId = "applicationId";
     private final Metrics metrics = new Metrics();
-    private final StreamsMetrics streamsMetrics = new MockStreamsMetrics(metrics);
+    private final StreamsMetrics streamsMetrics = new StreamsMetricsImpl(metrics, "mock-stream-metrics",
```

Comment:
As we mentioned in the ticket, the only test case that requires the underlying `metrics` object is `testMetrics`. For this purpose we should just rewrite this test function, to not reuse the class field `task` which is relying on the `MockStreamsMetrics`, but create another task that does pass in a real `StreamsMetricsImpl`.